WordPress/openverse-catalog
# Add `DEPLOYMENT.md` & deployment-related files

## Fixes


Fixes #672 by @AetherUnbound
Fixes #681 by @AetherUnbound

## Description


This PR adds a `DEPLOYMENT.md` file (modeled after [an analogous file in the frontend repo](https://github.com/WordPress/openverse-frontend/blob/4703ef7bcda2f7854ce392f9d864f9561a48c303/DEPLOYMENT.md)) which describes the current deployment setup & workflow for both Airflow and Postgres.

I've also moved the `dag-sync.sh` script from the EC2 initialization script in the infrastructure repo into this repo. This makes the deployment process clearer for community contributors and allows us to simplify the EC2 init. I've made some changes to the `dag-sync.sh` script in order to make it more generic and less dependent on our specific Terraform setup.

## Testing Instructions

Please read the documentation and let me know if anything is confusing or if I missed something :slightly_smiling_face:

You can also try running the DAG sync script with `./dag-sync.sh`. Nothing should happen if you just checked out the branch, but if you comment out the following line you should get a warning about a Slack URL not being supplied:

https://github.com/WordPress/openverse-catalog/blob/5fce606112072f39ac0f2693ebf38064b5f9f056/dag-sync.sh#L22

You can add gibberish to the call (e.g. `./dag-sync.sh will-not-work`) and `curl` should throw a fit about not being able to reach `will-not-work`.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
